src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs

```diff
@@ -136,12 +146,14 @@ private void WriteResources()
             var accessorNamespace = String.IsNullOrEmpty(ResourcesNamespace) ? _srNamespace : ResourcesNamespace;
             var accessorClassName = String.IsNullOrEmpty(ResourcesClassName) ? _srClassName : ResourcesClassName;
 
+
             if (_targetLanguage == TargetLanguage.CSharp)
             {
                 _targetStream.WriteLine($"namespace {accessorNamespace}");
                 _targetStream.WriteLine("{");
                 _targetStream.WriteLine("");
-                _targetStream.WriteLine($"    internal static partial class {accessorClassName}");
+                var accessor = AsEnum ? "enum" : "partial class";
+                _targetStream.WriteLine($"    internal static {accessor} {accessorClassName}");
```

Comment:
Is ‘static enum’ valid syntax?  As written this will produce ‘internal static enum foo’ when AsEnum is true.